app/src/main/java/org/oppia/android/app/hintsandsolution/SolutionViewModel.kt

```diff
@@ -1,28 +1,266 @@
-  val solutionCanBeRevealed = ObservableField<Boolean>(false)
-
-  fun computeSolutionContentDescription(): String {
-    return solutionSummary.get()?.let {
-      CustomHtmlContentHandler.fromHtml(
-        it,
-        imageRetriever = null,
-        customTagHandlers = mapOf()
-      ).toString()
-    } ?: ""
+/**
+ * [HintsAndSolutionItemViewModel] that represents a solution that the user may reveal.
+ *
+ * Instances of this class are created using its [Factory].
+ *
+ * @property solutionSummary the solution's explanation text (which may contain HTML)
+ * @property isSolutionRevealed whether the solution is currently expanded and viewable
+ */
+class SolutionViewModel private constructor(
+  val solutionSummary: String,
+  private val correctAnswer: InteractionObject,
+  val isSolutionRevealed: ObservableBoolean,
+  isSolutionExclusive: Boolean,
+  private val interaction: Interaction,
+  private val writtenTranslationContext: WrittenTranslationContext,
+  private val appLanguageResourceHandler: AppLanguageResourceHandler,
+  private val mathExpressionAccessibilityUtil: MathExpressionAccessibilityUtil
+) : HintsAndSolutionItemViewModel() {
+  /**
+   * A screenreader-friendly version of [solutionSummary] that should be used for readout, in place
+   * of the original summary.
+   */
+  val solutionSummaryContentDescription by lazy {
+    CustomHtmlContentHandler.fromHtml(
+      solutionSummary,
+      imageRetriever = null,
+      customTagHandlers = mapOf()
+    ).toString()
+  }
+
+  /** A displayable HTML representation of the correct answer presented by this model's solution. */
+  val correctAnswerHtml: String by lazy { computeCorrectAnswerHtml() }
+
+  /** A screenreader-friendly readable version of [correctAnswerHtml]. */
+  val correctAnswerContentDescription: String by lazy { computeCorrectAnswerContentDescription() }
+
+  private val correctAnswerTextStringResId = if (isSolutionExclusive) {
+    R.string.hints_list_exclusive_solution_text
+  } else R.string.hints_list_possible_solution_text
+
+  private fun computeCorrectAnswerHtml(): String {
+    val answerTextHtml = when (correctAnswer.objectTypeCase) {
+      NORMALIZED_STRING -> correctAnswer.normalizedString
+      REAL -> correctAnswer.real.toSimplifiedPlainString()
+      FRACTION -> correctAnswer.fraction.toAnswerString()
+      RATIO_EXPRESSION -> correctAnswer.ratioExpression.toAnswerString()
+      MATH_EXPRESSION -> when (interaction.id) {
+        "NumericExpressionInput" -> correctAnswer.mathExpression.toNumericExpressionHtml()
+        "AlgebraicExpressionInput" -> correctAnswer.mathExpression.toAlgebraicExpressionHtml()
+        "MathEquationInput" -> correctAnswer.mathExpression.toAlgebraicEquationHtml()
+        else -> error("Interaction ID not valid for math expressions: ${interaction.id}.")
+      }
+      LIST_OF_SETS_OF_TRANSLATABLE_HTML_CONTENT_IDS, SIGNED_INT, NON_NEGATIVE_INT,
+      TRANSLATABLE_SET_OF_NORMALIZED_STRING, TRANSLATABLE_HTML_CONTENT_ID,
```

Comment:
Does this work correctly for checkbox (item selection) answers? (answer would be displayed as a bullet list of html strings)

Reply:
Multiple choice & item selection aren't supported for solutions per web, same with image click. Drag & drop is, but I've intentionally left it out to help manage complexity (& I now audit imported lessons to make sure there are no such solutions; if that changes in the future, we'll have to finish adding support on Android in the same way as any of the other interactions).